src/pages/task/Logs.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useState } from "react";
+import {
+  GET_TASK_LOGS,
+  TaskLogsQuery,
+  TaskEventLogEntry,
+  LogMessage
+} from "../../gql/queries/get-task-logs";
+import { Radio, RadioGroup } from "@leafygreen-ui/radio-group";
+import { useLocation, useHistory, useParams } from "react-router-dom";
+import { useQuery } from "@apollo/react-hooks";
+import queryString from "query-string";
+import styled from "@emotion/styled/macro";
+import { TaskEventLogLine } from "./logs/TaskEventLogLine";
+import { LogMessageLine } from "./logs/LogMessageLine";
+
+enum LogTypes {
+  Agent = "agent",
+  System = "system",
+  Task = "task",
+  Event = "event"
+}
+const DEFAULT_LOG_TYPE = LogTypes.Task;
+enum QueryParams {
+  LogType = "logtype"
+}
+
+const renderLogs = ({
+  currentLog,
+  eventLogEntryData,
+  logMessageData
+}: {
+  currentLog: LogTypes;
+  eventLogEntryData: [TaskEventLogEntry];
+  logMessageData: [LogMessage];
+}): JSX.Element => {
+  const noLogs = <div id="cy-no-logs">No logs</div>;
+  if (currentLog === LogTypes.Event) {
+    if (!eventLogEntryData.length) {
+      return noLogs;
+    }
+    return (
+      <>
+        {eventLogEntryData.map((data, i) => (
+          <TaskEventLogLine key={i} {...data} />
+        ))}
+      </>
+    );
+  }
+  if (!logMessageData || !logMessageData.length) {
+    return noLogs;
+  }
+  return (
+    <>
+      {logMessageData.map((data, i) => (
+        <LogMessageLine key={i} {...data} />
+      ))}
+    </>
+  );
+};
+
+export const Logs: React.FC = props => {
+  const { id } = useParams<{ id: string }>();
+  const { search, pathname } = useLocation();
+  const { replace } = useHistory();
+  const [currentLog, setCurrentLog] = useState<LogTypes>(DEFAULT_LOG_TYPE);
+  const { data, loading, error } = useQuery<TaskLogsQuery>(GET_TASK_LOGS, {
+    variables: { id }
+  });
+  const parsed = queryString.parse(search);
+  const logTypeParam = (parsed[QueryParams.LogType] || "")
+    .toString()
+    .toLowerCase();
+  useEffect(() => {
```

Comment:
when i see a `useEffect` like this, my question is "what is this doing?" i read it a couple times and still am not sure. if i were to make this into a custom hook then i would put a comment with docs for the hook to explain what it's accomplishing. wondering how we can make the utility of our hooks more explicit.